### PR TITLE
Use inline functions for errno checks

### PR DESCRIPTION
**Pull request by pando-emil**
Since winsock do not use errno, and have different error codes, this is needed to be windows compatible

This pull request is a split of PR #1061

### CODE REVIEW

File: src/apps/common/apputils.c [hunk 1]
```diff
@@ -707,61 +707,57 @@ int get_socket_mtu(evutil_socket_t fd, int family, int verbose) {
...
-  case EACCES:
+  }
+  if (socket_eacces()) {
     /* Permission denied.
      * Just ignore, we might be blocked
      * by some firewall policy. Try again
      * and hope for the best.
      */
     return 1;
-  default:
```
[the remark this answers is not in the document]
Comment (pando-emil) [reply src/apps/common/apputils.c:760]: It is not a switch anymore. The same "catch rest" is present below the if statements.

[the remark this answers is not in the document]
Comment (KangLin) [reply src/apps/common/apputils.c:760]: O. but I think it's better to read with the switch.

[the remark this answers is not in the document]
Comment (KangLin) [reply src/apps/common/apputils.c:760]: Performance becomes lower with if. Although it is not much reflected here

**Review by KangLin — APPROVED** [review]

File: src/apps/common/apputils.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (eakraly) [reply src/apps/common/apputils.c:760]: Any reason not to have `else if`?

[the remark this answers is not in the document]
Comment (pando-emil) [reply src/apps/common/apputils.c:760]: @KangLin I agree with the readability on a switch, but with the inline functions it was harder to achieve a switch...

[the remark this answers is not in the document]
Comment (KangLin) [reply src/apps/common/apputils.c:760]: ok. keep your code

**Comment by KangLin** [comment]
@eakraly Please merge

**Review by eakraly — APPROVED** [review]

**Pull request merged**